### DISCUSSION TITLE
`contex.query` types should infer from the GraphQL types

### DISCUSSION THREAD

**Discussion by tilman** [keystonejs/keystone#8496] [Feature requests]
currently the return type of all operations (like findOne, findMany...) uses `any` a lot, specifically in the most interesting parts of the returned data. 

It would be better if the type could be inferred from the query (like in prisma or trpc for example). Therefore It would be necessary to change the query from a string to an object which describes the data. Instead of this
`query: 'id name about { document }'`
write this
`query: {id:true, name: true, about: { document: true }` or something similar.

This approach would have two benefits over an type generator approach (generate types from query string with some script automatically and change findOne to accept an type-generic):
1.) "cmd+click code jumping" works out of the box
2.) No setup of another tool needed (the type generator)

The benefits of input & output types are obvious:
1.) code completion and type checking (validation) for the input query
2.) code completion for work based on the return type

**Comment by dcousens** [comment]
Marking this as a duplicate of https://github.com/keystonejs/keystone/discussions/8498, @tilman if you could follow up with any feedback on our proposal, that'd be great :yellow_heart:
